tests/testbed_validation_rules.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABC, abstractmethod
+import re
+import logging
+from pyats.topology.testbed import Testbed
+
+
+_log = logging.getLogger(__name__)
+
+
+class Rule(ABC):
+
+    ip_pattern = r"^\d{3}\.\d{3}\.\d{1,3}\.\d{1,3}$"
+
+    @abstractmethod
+    def is_passed(self, testbed: Testbed) -> bool:
+        """Provides a result of rule execution"""
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        """Returns rule name"""
+        pass
+
+
+class CheckUserCredentials(Rule):
+    def is_passed(self, testbed: Testbed) -> bool:
+        for device in testbed:
+            device.connect()
+            assert device.connected
+        return True
+
+    def __str__(self):
+        return "Check user credentials"
+
+
+class CheckDevicesIp(Rule):
+    def is_passed(self, testbed: Testbed) -> bool:
+        for device in testbed:
+            device_ip = str(device.connections.main.ip)
+            assert re.fullmatch(self.ip_pattern, device_ip)
+        return True
+
+    def __str__(self):
+        return "Check devices ip format"
+
+
+class CheckSeleniumGridUrl(Rule):
+
+    url_pattern = r"^https?:\/\/.+:\d+$"
+
+    def is_passed(self, testbed: Testbed) -> bool:
+        grid_url = testbed.custom["selenium-grid"]
+        assert re.fullmatch(self.url_pattern, grid_url)
+        return True
+
+    def __str__(self):
+        return "Check Selenium grid url format"
+
+
+class CheckServerIp(Rule):
+    def is_passed(self, testbed: Testbed) -> bool:
+        server_ip = testbed.servers.server_alias.address
+        assert re.fullmatch(self.ip_pattern, server_ip)
+        return True
+
+    def __str__(self):
+        return "Check server ip format"
+
+
+class Rules:
+    def __init__(self, set_of_rules):
```

Comment:
@SergiyBeno just use simple names and don't forget about type hints
```suggestion
    def __init__(self, rules: Sequence[Rule]):
```

Reply:
Sure, my bad. Fixed.